libs/core/langchain_core/runnables/config.py

```diff
@@ -131,17 +123,27 @@ def _set_config_context(config: RunnableConfig) -> None:
     Args:
         config (RunnableConfig): The config to set.
     """
-    from langsmith import (
-        RunTree,  # type: ignore
-        run_helpers,  # type: ignore
-    )
+    from langchain_core.tracers.langchain import LangChainTracer
 
     var_child_runnable_config.set(config)
-    if hasattr(RunTree, "from_runnable_config"):
-        # import _set_tracing_context, get_tracing_context
-        rt = RunTree.from_runnable_config(dict(config))
-        tc = run_helpers.get_tracing_context()
-        run_helpers._set_tracing_context({**tc, "parent": rt})
+    if (
+        (cb := config.get("callbacks"))
+        and (prid := getattr(cb, "parent_run_id", None))  # Is callback manager
```

Comment:
I'd recommend keeping the same variable name as the name that we're looking up to keep the mental burden low.

`prid` is faster to read, but harder to read since now i have to think about what this variable name means

prid -> parent_id

t -> handler 

etc